src/seedu/addressbook/Utils.java

```diff
@@ -0,0 +1,51 @@
+package seedu.addressbook;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Utility methods
+ */
+public class Utils {
+
+    /**
+     * Throws an assertion error if any of the given arguments is null.
+     */
+    public static void assertNotNull(Object... items) {
+        int argIndex = 0;
+        for (Object item : items) {
+            if (Objects.isNull(item)) {
+                throw new AssertionError("Argument at index " + argIndex + " is null");
+            }
+            argIndex++;
+        }
+    }
```

Comment:
In terms of modularity, each data type class (`Person`, `Name`, `UniqueTagList`) etc has certain constraints they must fulfill. 

So if we see a random `Name` object, we know that it conforms to the naming format, if we see a random `UniqueTagList` object, we know that it contains no duplicates etc..

Is there some nicer way of working in a not-null guarantee? I'd really like instances of data classes to behave as expected/specified in every possible case.